components/Button.tsx

```diff
@@ -1,33 +1,58 @@
 import React from 'react';
-import { Text, TouchableOpacity, TouchableOpacityProps } from 'react-native';
+import {
+  Text,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  TextStyle,
+  Image,
+  ImageSourcePropType,
+  View,
+} from 'react-native';
 
 interface ButtonProps extends TouchableOpacityProps {
   label: string;
-  size?: string; 
-  color?: string; 
+  size?: string;
+  color?: string;
   className?: string;
   textClassName?: string;
+  textStyle?: TextStyle;
+  imgSrc?: ImageSourcePropType;
```

Comment:
Instead of `textStyle`, let's add a `poppinsVariant` prop to define the font variant for the button. This allows us to map the variants to specific Poppins font styles, which enforces consistency and simplifies usage.

In the future, when we integrate Tailwind configuration for fonts, we can transition later.

```suggestion
  poppinsVariant?: 'bold' | 'medium' | 'regular' | 'light';
```